tests/transformation/test_ast_transformers.py

```diff
@@ -0,0 +1,162 @@
+import enum
+import unittest
+from dataclasses import dataclass, field
+from typing import List
+
+from pylasu.model import Node
+from pylasu.transformation.transformation import ASTTransformer, PropertyRef, ast_transformer
+
+
+@dataclass
+class CU(Node):
+    statements: List[Node] = field(default_factory=list)
+
+
+@dataclass
+class DisplayIntStatement(Node):
+    value: int = 0
+
+
+@dataclass
+class SetStatement(Node):
+    variable: str = ""
+    value: int = 0
+
+
+class Operator(enum.Enum):
+    PLUS = '+'
+    MULT = '*'
+
+
+class Expression(Node):
+    pass
+
+
+@dataclass
+class IntLiteral(Expression):
+    value: int
+
+
+@dataclass
+class GenericBinaryExpression(Node):
+    operator: Operator
+    left: Expression
+    right: Expression
+
+
+@dataclass
+class Mult(Node):
+    left: Expression
+    right: Expression
+
+
+@dataclass
+class Sum(Node):
+    left: Expression
+    right: Expression
+
+
+class ASTTransformerTest(unittest.TestCase):
+
+    def test_identitiy_transformer(self):
+        prop = PropertyRef("statements")
+        transformer = ASTTransformer()
+        transformer.register_node_factory(CU, CU).with_child(prop, prop)
```

Comment:
Is this different from an IdentityTransformer?

Reply:
Yes because it has a child. This is ported as-is from Kolasu.